### PR TITLE
Aswathy/Terms use validation

### PR DESCRIPTION
**Pull request by aswathy-deriv**
Changes:

-   ...

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [x] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/pages/signup-affiliates/components/_account-terms.tsx [hunk 1]
```diff
@@ -1,14 +1,51 @@
...
+type ConsentProps = {
+    non_pep_declaration: boolean
+    tnc_accepted: boolean
+    is_brokers_checked: boolean
+    is_eu_checked: boolean
+    is_partner_checked: boolean
+}
+
+type AccountTermsProps = {
+    updatedData: (e) => void
```
Comment (michio-rework) [inline src/pages/signup-affiliates/components/_account-terms.tsx:16]:
```sugesstion
updateData: (configs: ConsentProps) : void
```
we can change `ConsentProps` to `TAffiliateTerms`, just sugession though :pray:

File: src/pages/signup-affiliates/components/_account-terms.tsx [hunk 2]
```diff
@@ -1,14 +1,51 @@
...
+    onValidate: (e) => void
```
Comment (michio-rework) [inline src/pages/signup-affiliates/components/_account-terms.tsx:17]: onValidate: (validated : boolean) => void

File: src/pages/signup-affiliates/components/_account-terms.tsx (hunk 1 above)
Comment (aswathy-deriv) [reply src/pages/signup-affiliates/components/_account-terms.tsx:16]: okay

File: src/pages/signup-affiliates/components/_account-terms.tsx (hunk 2 above)
Comment (aswathy-deriv) [reply src/pages/signup-affiliates/components/_account-terms.tsx:17]: okay

**Review by habib-deriv — APPROVED** [review]
LGTM

**Pull request merged**